src/redux/stores/balances/balancesHelpers.ts

```diff
@@ -54,21 +54,45 @@ export const getOwnedTokenIdsOfWallet = async (
     const transferFilter = collectionContract.filters.Transfer(null, walletAddress);
 
     const events: Event[] = await collectionContract.queryFilter(transferFilter, 0);
+
     /* get token ids from past events */
     const foundTokenIds: BigNumber[] = events.map(e => e.args?.at(2));
 
-    return getUniqueTokensForWallet(foundTokenIds, collectionContract, walletAddress);
+    /* get unique values */
+    const uniqueTokenIds = [...new Set(foundTokenIds)];
+
+    /* get only the owned token ids */
+    const ownedTokenIds = uniqueTokenIds.filter(async id => {
+      const addr = await collectionContract.ownerOf(id);
+      return addr === walletAddress;
+    });
```

Comment:
This filter is causing the issue.

The owner is retrieved and even if the comparison returns false when checking, the array is still the same after this filter. 🤔

Reply:
Aha. It's because you can't put an aysnc function as a filter. You should do something like this:

```
const tokenOwners: string[] = await Promise.all(uniqueTokenIds.map(tokenId => collectionContract.ownerOf(tokenId)));
    const ownedTokenIds = uniqueTokenIds
      .filter((id, index) => tokenOwners[index] === walletAddress);
```